internal/cli/profile_populate.go

```diff
@@ -0,0 +1,176 @@
+// Copyright 2018 CoreOS Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cli
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/coreos/torcx/internal/torcx"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+)
+
+var (
+	cmdProfilePopulate = &cobra.Command{
+		Use:   "populate",
+		Short: "Populates the store with all image required by the given profile",
+		RunE:  runProfilePopulate,
+	}
+
+	flagProfilePopulateName      string
+	flagProfilePopulatePath      string
+	flagProfilePopulateOsVersion string
+)
+
+func init() {
+	cmdProfile.AddCommand(cmdProfilePopulate)
+	cmdProfilePopulate.Flags().StringVar(&flagProfilePopulateName, "name", "", "profile name to populate")
+	cmdProfilePopulate.Flags().StringVar(&flagProfilePopulatePath, "file", "", "profile file to populate")
+	cmdProfilePopulate.Flags().StringVarP(&flagProfilePopulateOsVersion, "os-release", "n", "", "override OS version")
+}
+
+func runProfilePopulate(cmd *cobra.Command, args []string) error {
+	ctx, cancel := context.WithTimeout(context.Background(), 12*time.Hour)
+	defer cancel()
+
+	commonCfg, err := fillCommonRuntime(flagProfilePopulateOsVersion)
+	if err != nil {
+		return errors.Wrap(err, "common configuration failed")
+	}
+
+	if len(args) != 0 {
+		return cmd.Usage()
+	}
+
+	if flagProfilePopulatePath == "" {
+		if flagProfilePopulateName == "" {
+			flagProfilePopulateName, err = commonCfg.NextProfileName()
+			if err != nil {
+				return errors.Wrapf(err, "unable to determine next profile")
+			}
+
+			logrus.Infof("using next profile %q", flagProfilePopulateName)
+
+			if flagProfilePopulateName == torcx.VendorProfileName {
+				logrus.Warn("using default profile (%s)", torcx.VendorProfileName)
```

Comment:
might want to say "populating the default profile has no effect" or something explaining why this is a warning.